### ISSUE TITLE
Cannot find resolvers for arraysOfReference when located within object

### ISSUE THREAD

**Issue by mikesir87** [soulwing/prospecto#28]
In PR #27, the first commit is for what you added and it works.  Look at the second commit (5988d7ab56170e889d8031964ae80924e05f6361) to test this issue.

When running, I get the following exception:

```
Exception in thread "main" org.soulwing.prospecto.api.ModelEditorException: org.soulwing.prospecto.runtime.reference.ReferenceResolverNotFoundException: cannot find a resolver for reference of type org.soulwing.prospecto.tests.ArrayOfReferencesDemo$ReferencedObject
	at org.soulwing.prospecto.runtime.entity.ConcreteMutableViewEntity.inject(ConcreteMutableViewEntity.java:111)
	at org.soulwing.prospecto.api.association.AbstractAssociationManager.newAssociate(AbstractAssociationManager.java:50)
	at org.soulwing.prospecto.runtime.association.ConcreteToOneAssociationUpdater.update(ConcreteToOneAssociationUpdater.java:81)
	at org.soulwing.prospecto.runtime.node.ObjectNode.inject(ObjectNode.java:92)
	at org.soulwing.prospecto.runtime.entity.ConcreteMutableViewEntity.inject(ConcreteMutableViewEntity.java:119)
	at org.soulwing.prospecto.runtime.node.RootObjectNode.create(RootObjectNode.java:62)
	at org.soulwing.prospecto.runtime.editor.ConcreteModelEditor.create(ConcreteModelEditor.java:58)
	at org.soulwing.prospecto.tests.ArrayOfReferencesDemo.main(ArrayOfReferencesDemo.java:160)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:144)
Caused by: org.soulwing.prospecto.runtime.reference.ReferenceResolverNotFoundException: cannot find a resolver for reference of type org.soulwing.prospecto.tests.ArrayOfReferencesDemo$ReferencedObject
	at org.soulwing.prospecto.runtime.reference.LinkedListReferenceResolverService.resolve(LinkedListReferenceResolverService.java:43)
	at org.soulwing.prospecto.runtime.node.ArrayOfReferenceNode.inject(ArrayOfReferenceNode.java:55)
	at org.soulwing.prospecto.runtime.entity.ConcreteMutableViewEntity.inject(ConcreteMutableViewEntity.java:119)
	at org.soulwing.prospecto.runtime.entity.ConcreteMutableViewEntity.inject(ConcreteMutableViewEntity.java:108)
	... 12 more
```

When setting a breakpoint in ```LinkedListReferenceResolverService```, the collection of resolvers is empty.

**Comment by ceharris** [comment]
Interesting. I do essentially the same test in `EditorTest.testObjectValueArrayOfReferencesListValue` and the resolver is found.

**Comment by ceharris** [comment]
Now I understand... what you meant to say in the description was "Cannot find resolver when arrayOfReferences is used in **nested** object"

**Comment by ceharris** [comment]
This will be resolved by #29
